src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Serialization.cs

```diff
@@ -12,13 +13,13 @@ namespace System.Text.Json
     internal static partial class ThrowHelper
     {
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void ThrowArgumentException_DeserializeWrongType(Type type, object value)
+        public static void ThrowArgumentException_DeserializeWrongType(Type? type, object value)
```

Comment:
Same here. Can we annotate **all** the `Throw...` methods with `[DoesNotReturn]`

`ThrowArgumentException_DeserializeWrongType`
`ThrowInvalidOperationException_SerializerCycleDetected`
`ThrowJsonException_DeserializeUnableToConvertValue` overloads
`ThrowJsonException_DepthTooLarge`
`ThrowJsonException_SerializationConverterRead`
`ThrowJsonException_SerializationConverterWrite`
`ThrowJsonException`
`ThrowInvalidOperationException_SerializationConverterNotCompatible`
`ThrowInvalidOperationException_SerializationConverterOnAttributeInvalid`
`ThrowInvalidOperationException_SerializationConverterOnAttributeNotCompatible`
`ThrowInvalidOperationException_SerializerOptionsImmutable`
`ThrowInvalidOperationException_SerializerPropertyNameConflict`
`ThrowInvalidOperationException_SerializerPropertyNameNull`
`ThrowInvalidOperationException_SerializerDictionaryKeyNull`
`ReThrowWithPath` overloads
`ThrowInvalidOperationException_SerializationDuplicateAttribute`
`ThrowInvalidOperationException_SerializationDuplicateTypeAttribute`
`ThrowInvalidOperationException_SerializationDataExtensionPropertyInvalid`
`ThrowNotSupportedException_DeserializeCreateObjectDelegateIsNull`

Reply:
sure